akka-http-core/src/test/scala/akka/http/TestServer.scala

```diff
@@ -45,4 +64,16 @@ object TestServer extends App {
          |    </ul>
          |  </body>
          |</html>""".stripMargin))
+
+  def echoWebsocketService: Flow[Message, Message, Unit] =
+    Flow[Message] // just let message flow directly to the output
+      .via(akka.http.util.printEvent("echoWebsocketService"))
+
+  def greeterWebsocketService: Flow[Message, Message, Unit] =
+    Flow[Message]
+      .collect {
+        case TextMessage.Strict(name)         ⇒ TextMessage.Strict(s"Hello '$name'")
+        case TextMessage.Streamed(nameStream) ⇒ TextMessage.Streamed(Source.single("Hello ") ++ nameStream mapMaterialized (_ ⇒ ()))
```

Comment:
I thought you are using mapMaterialized(_ ⇒ ()) because of the concat operator expecting Unit